lib/auth/github.go

```diff
@@ -372,41 +375,74 @@ type orgResponse struct {
 	Login string `json:"login"`
 }
 
-// getTeams retrieves a list of teams authenticated user belongs to
+// getTeams retrieves a list of teams authenticated user belongs to.
 func (c *githubAPIClient) getTeams() ([]teamResponse, error) {
-	bytes, err := c.get("/user/teams")
+	var result []teamResponse
+
+	bytes, nextPage, err := c.get("/user/teams")
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+
+	// Extract the first page of results and append them to the full result set.
 	var teams []teamResponse
 	err = json.Unmarshal(bytes, &teams)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	return teams, nil
+	result = append(result, teams...)
+
+	// If the response returned a next page link, continue following the next
+	// page links until all teams have been retrieved.
+	for nextPage != "" {
```

Comment:
can you limit the number of pages to some reasonably high number? E.g. if github API goes crazy and will start returning circular link, this logic will hang forever.